src/NHibernate.Validator/Util/AttributeUtils.cs

```diff
@@ -14,8 +16,48 @@ public static class AttributeUtils
 		/// <returns></returns>
 		public static bool AttributeAllowsMultiple(Attribute attribute)
 		{
-			Attribute usageAttribute = Attribute.GetCustomAttribute(attribute.GetType(), typeof(AttributeUsageAttribute));
+			var usageAttribute = Attribute.GetCustomAttribute(attribute.GetType(), typeof(AttributeUsageAttribute));
 			return ((AttributeUsageAttribute)usageAttribute).AllowMultiple;
 		}
+
+		/// <summary>
+		/// Returns true if the attribute can be declared more than one time for the same element and this attribute marked with [AllowsMultipleWithSameTags]
+		/// </summary>
+		/// <param name="attribute"></param>
+		/// <returns></returns>
+		public static bool AttributeAllowsMultipleWithIntersectingTags(Attribute attribute)
+		{
+			var attr = Attribute.GetCustomAttribute(attribute.GetType(), typeof(AllowsMultipleWithIntersectingTagsAttribute));
+			return AttributeAllowsMultiple(attribute) && attr != null;

```

Comment:
Revert condition order: when using short-circuiting boolean operators, the less costly operation should be done first.